src/sqliteformat.cpp

```diff
@@ -2204,8 +2204,8 @@ QDateTime SqliteFormat::fromOriginTime(sqlite3_int64 seconds, const QByteArray &
         const QTimeZone timezone(zonename);
         if (timezone.isValid()) {
             dt = fromOriginTime(seconds).toTimeZone(timezone);
-        } else if (d->mTimeZone.isValid() && d->mTimeZone.id() == zonename) {
```

Comment:
Pondering should we keep the behavior by passing mCalendar from SqliteStorage to SqliteFormat and using here something like mCalendar->timeZone(). Should work more or less how it used to without needing to pass and maintain a separate copy of the timezone?

Reply:
Btw, without this, the above "First try system zones" comment isn't necessarily the best description if that's the only thing attempted.

Reply:
Well, passing mCalendar to SqliteFormat just to get it there would complicate the code much to keep a branching that may not be used. Why I think so:
KCalendarCore::Calendar can only have QTimeZone associated. And I'm assuming that QTimeZone(tz->id()) is always succesfull if tz a valid QTimeZone. In other words, when QTimeZone("foo").isValid() is false, it cannot validate the test (mCalendar->timeZone().id() == "foo") which is basically the branching that has been removed.

This branching existed before because KTimeZone allowed to have user defined timezones, which means that KTimeZone("foo") may be invalid, but KTimeZone().setId("foo") and then defining transitions… was possible. As far as I know, QTimeZone has no setters.

All that being said, if we keep removing this branching, I need to adjust the comment as you mentioned.

Reply:
Ok, with the latest version at least trying to handle the time I'm confident enough.